Nos/Views/CircularFollowButton.swift

```diff
@@ -14,10 +14,15 @@ struct CircularFollowButton: View {
 
     private let diameter: CGFloat = 30
 
+    @State var disabled = false
+
     var body: some View {
         let following = currentUser.isFollowing(author: author)
 
         Button {
+            disabled = true
+            defer { disabled = false }
```

Comment:
I don't think this `defer` will execute at the correct time. It will get run immediately after `Task` starts, not after it completes. I think you need to move `disabled = false` into the task. You could also `await` the tasks completion like `_ = await task.result` but that would require `Button` to accept an `async` closure, which it doesn't.

Reply:
Oh, oops! Thanks!